nautobot/utilities/filters.py

```diff
@@ -411,6 +440,8 @@ def get_filters(cls):
                 new_filters[new_filter_name] = new_filter
 
         filters.update(new_filters)
+        # TODO: Verify that this brute force method is the best way to ensure pluggable filters gets updated
+        cls.base_filters = filters
```

Comment:
There is no instance of the class created, I could be missing something, but I don't see how this would work.

I have changed this up, I have some environment issues so I haven't tested it, but in theory the option of re-assigning base_filters only on registration should work and to your point, not introduce unexpected behavior.